File: spring-boot-cloud/food-nutrients/src/main/java/com/recipemanager/foodnutrients/entity/Nutrient.java
package com.recipemanager.foodnutrients.entity;


import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

@Getter
@Setter
@Entity
@Table(name = "nutrient")
public class Nutrient {
    @Id
    private Long Id;
    private String name;
    private String nameFr;
    private String symbol;
    private String friendlyNameFr;
    private String unit;
    private Float requirement;

    @ManyToOne
    @OnDelete(action = OnDeleteAction.CASCADE)
    @JoinColumn(name = "nutrient_group_id")
    private NutrientGroup nutrientGroup;

}

File: angular-front/src/app/recipes/models/Nutrient.ts
export interface NutrientGroup {
  id: number,
  name: string,
  nameFr: string
}
export interface Nutrient {
  id: number,
  name: string,
  nameFr: string,
  friendlyNameFr: string,
  symbol: string,
  requirement: number,
  unit: string,
  nutrientGroup: NutrientGroup

}
